src/components/ZoneFilterPanel.js

```diff
@@ -74,19 +74,29 @@ export function ZoneFilterPanel({ onFilterSelect, autoFocus }) {
                   marginLeft: 10,
                 }}
                 onChange={e => {
+                  // @todo #223 show dropdown or "autocomplete" for province name?
                   const { value } = e.target
                   setState({ ...state, value: value })
                 }}
                 onBlur={e => {
                   setState({ ...state, value: state.query })
                 }}
                 onKeyPress={e => {
-                  if (e.key == "Enter" && isProvinceExist(state.value)) {
-                    trackEvent("Search for province")
-                    setState(
-                      { ...state, query: state.value },
-                      navigate(filterPath(state.value))
-                    )
+                  const { value } = state
+                  if (e.key == "Enter") {
+                    if (isProvinceExist(state.value)) {
+                      trackEvent("Search for province")
+                      setState(
+                        { ...state, query: state.value },
+                        navigate(filterPath(state.value))
+                      )
+                    } else {
+                      // @todo #223 clear input box
+                      alert(
+                        "ไม่พบจังหวัดที่ท่านระบุ กรุณาตรวจสอบชื่อจังหวัดทีท่านกรอกอีกครั้ง"
+                      )
+                      setState({ ...state, value: "" })
```

Comment:
Doesn't setState clear the input box?

Reply:
I try to use the existing value, but the error page is shown.

It seems like, if I set the value to the current (incorrect province's name) one, it tries to search non-exist province and show the following error:

![image](https://user-images.githubusercontent.com/3356814/54901636-a9962480-4f09-11e9-90a0-f4772676bf0e.png)

So, for now, I clear it out to prevent the problem.

Reply:
yep. The function component `renderFilter(filterName, label="")` require `filterName` to be valid otherwise it will crash. Also, it goes back to the previous successful query due to `onBlur` (since we cannot leave renderFilter to be invalid).

Reply:
but it seems strange since I've already used `state.query` not `state.value`...